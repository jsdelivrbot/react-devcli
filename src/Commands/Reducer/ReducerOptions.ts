export const name = 'reducer';

export const description = 'Reducer description';

export const flags = {};
