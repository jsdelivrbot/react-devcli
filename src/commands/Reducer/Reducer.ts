import {Arguments} from 'yargs';
import ACommand from '../Command/ACommand';
import {IMakeable} from '../Command/contracts/IMakeable';
import {IRemovable} from '../Command/contracts/IRemovable';
import {ITestable} from '../Command/contracts/ITestable';

class Reducer extends ACommand implements IRemovable, ITestable, IMakeable {
    constructor() {
        const name = 'reducer';
        const description = 'Reducer description';
        const options = {
        };

        super(name, description, options);
    }

    public run(argv: Arguments): void {
        console.log(argv);
    }

    public help(argv: Arguments): void {
        console.log(argv);
    }

    public remove(argv: Arguments): void {
        console.log(argv);
    }

    public test(argv: Arguments): void {
        console.log(argv);
    }

    public make(argv: Arguments): void {
        console.log(argv);
    }
}

export default new Reducer();
