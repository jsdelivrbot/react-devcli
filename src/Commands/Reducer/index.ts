import Reducer from './Reducer';

export default new Reducer();
