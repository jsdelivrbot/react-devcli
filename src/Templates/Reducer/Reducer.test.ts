import {js, ts} from './Reducer';

describe('Reducer template test suite', (): void => {
    test('should match reducer template snapshot for JS language', (): void => {
        expect(js()).toMatchSnapshot();
    });

    test('should match reducer template snapshot for TS language', (): void => {
        expect(ts()).toMatchSnapshot();
    });
});
