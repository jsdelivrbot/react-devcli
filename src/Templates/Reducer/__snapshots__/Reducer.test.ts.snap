// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer template test suite should match reducer template snapshot for JS language 1`] = `""`;

exports[`Reducer template test suite should match reducer template snapshot for TS language 1`] = `""`;
